### PR TITLE
View

### PR DESCRIPTION
**Pull request by faisalakb**
In this project apply the following features : :100: 

 - Implement the design (views) for users and posts :+1:

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1 +1,29 @@
-<h1>Display all posts</h1>
\ No newline at end of file
+<img src="<%= asset_path('im12.jpg') %>" alt="User Photo">
```
Comment (V-Blaze) [inline app/views/posts/index.html.erb]: - You have done great so far :+1: But there are changes and requirements you still need to meet before moving to the next milestone. :+1: 

- [ ]  I noticed there's an issue with the migration order – the migration to add foreign keys assumes the existence of the "comments" table, which is created in a later migration.

To fix this, we should ensure that the migrations run in the correct order. Here's a quick guide:

**Update Migration Timestamps:**
Make sure the migration file **20231115151208_add_foreign_keys.rb** has a timestamp earlier than the migration that creates the "comments" and other tables. If not, consider updating the timestamp to an earlier date.."  :wave: :small_airplane: 

- - Screenshot reference:
Selection_040

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1 +1,13 @@
...
+<div class="container">
+  <% @users.each do |user| %>
+    <a href="<%= user_path(user.id) %>" class="userLink">
+      <img src="<%= asset_path('im12.jpg') %>" alt="User Photo">
+      <div class="userCon">
+        <h2><%= user.name %></h2>
+        <p>Number of posts: <%= user.posts.count %></p>
+      </div>
+    </a>
+  <% end %>
```
Comment (V-Blaze) [inline app/views/users/index.html.erb]: - [ ] According to the project requirement, your design layout is meant to look exactly to the wireframe, but looking at your current design, that seems to be the case. I'll suggest you update this to follow the design guide. :smile: :wave: 

- - Your current design:
Screenshot from 2023-11-24 08-45-48

- - Wireframe:
Screenshot from 2023-11-24 08-45-37

- [ ] Also the users' photo `" alt="User Photo">` is meant to come from the photo URL the user s=added while creating his/her account and not an image asset file. I'll suggest you update this too. :+1: :wave:

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,25 @@
...
+
+<% if @user.recent_posts.any? %>
+  <h2>Recent Posts:</h2>
+  <div class ="recPost">
+  <% @user.posts.each_with_index do |post, index| %>
+    <p><strong>Post#<%= index + 1 %>:</strong> <%= post.title %></p>
+    <p><strong>Post Detail:</strong> <%= post.text %></p>
+    <p><strong>Comments Count:</strong> <%= post.comments_counter %></p>
+    <p><strong>Likes Count:</strong> <%= post.likes_counter %></p>
+    <hr>
```
Comment (V-Blaze) [inline app/views/users/show.html.erb]: - [ ] According to the project requirement, you are meant to use the `recent_posts` method to return the 3 most recent posts and display them, currently, you are checking if they exist, but then go ahead to loop and display all the users post and that's not correct. You are meant to use the method to return the 3 posts and then loop and display them. :+1: :+1: 
```erb
<% @user.recent_posts do |post| %>
```

- [ ] Just as I pointed out above about following the given wireframe, you also need to update this.
- - Your current Design:
Screenshot from 2023-11-24 08-51-48

- - Wireframe:
Screenshot from 2023-11-24 08-51-59

- [ ] Also this `" alt="User Photo">` is not totally correct. The user's photo is meant to be from the user's photo link in the database and not a hardcoded image file. :+1: :100:

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,29 @@
...
+  <% post.recent_comments.each do |comment| %>
+            <% if comment.user.present? %>
+              <p><%= comment.user.name %>: <%= comment.text %></p>
+            <% else %>
+              <p>Anonymous: <%= comment.text %></p>
+            <% end %>
+          <% end %>
+  
+  </div>
+<% end %>
```
Comment (V-Blaze) [inline app/views/posts/index.html.erb]: - [ ] Just as I have suggested above, You need to make sure your design follows the given wireframe, Your current design doesn't follow the layout completely, I'll suggest you update this correctly. :+1: :100: 

- - Your current Design:
Selection_041

- - Wireframe:
Screenshot from 2023-11-24 09-26-10

- [ ] Also as I suggested above this `" alt="User Photo">` is not totally correct. The user's photo is meant to be from the user's photo link in the database and not a hardcoded image file. :+1: :100:

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1 +1,21 @@
...
+<div class="post-container">
+  <h2>Post #<%= @post.id %> by <%= @post.author.name %></h2>
+  <div class='userPost'>
+    <p><%= @post.text %></p>
+  </div>
+  <p class="coli">Comments: <%= @post.comments.count %>, Likes: <%= @post.likes.count %></p>
+</div>
+
+<h2>Comments</h2>
+<div class="comment-list">
```
Comment (V-Blaze) [inline app/views/posts/show.html.erb]: - [ ] Just as I have suggested above, You need to make sure your design follows the given wireframe, Your current design doesn't follow the layout completely, I'll suggest you update this correctly. :+1: :100: 

- - Your current Design:
Selection_042


- - Wireframe:
Screenshot from 2023-11-24 09-30-19

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @faisalakb 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Controller methods implemented correctly✔
- Created all view files👋 
-  Correct git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by faisalakb** [comment]
Please check i updated all the above comments

**Review by aamirkhan2478 — APPROVED** [review]
Hi @faisalakb , 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly :+1: 
✅ No linter error :+1: 
✅ Good commit messages :+1: 
✅ Good descriptive PR :+1: 
✅ Well done :clap: you did all changes given by the previous reviewer :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**